src/Observers/UrlRewriteObserver.php

```diff
@@ -417,6 +422,24 @@ protected function resolveCategoryIds($categoryId, $topLevel = false, $storeView
         }
     }
 
+    /**
+     * Check if category is a child of the root category
+     *
+     * @param array $category The category to check
+     *
+     * @return bool
+     */
+    protected function isChildOfRootCategory($category)
+    {
+        // split the path into its segments
+        $categoryPathParts = explode('/', $category[MemberNames::PATH]);
```

Comment:
If a category is not yet present but created on the fly by pacemaker `$category[MemberNames::PATH])` is empty and the method will return `false`.

Reply:
We somehow have to catch this case and find a different solution.

Reply:
I will improve the Steps to reproduce

Reply:
isn't this similar to https://github.com/techdivision/import/issues/209 ?